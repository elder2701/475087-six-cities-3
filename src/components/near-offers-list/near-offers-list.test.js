import React from "react";
import renderer from "react-test-renderer";
import NearOffersList from "./near-offers-list.jsx";

const offers = [
  {
    id: 1,
    figurePreview: `img/room.jpg`,
    mark: `Premium`,
    price: 822,
    rating: 4,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    allFigures: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    insideFeatures: [`Apartment`, `3 Bedrooms`, `Max 4 adults`],
    insideOptions: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
      `Cabel TV`,
      `Fridge`
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    user: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    status: `pro`,
    coordinates: [52.3909553943508, 4.85309666406198],
    comments: [
      {avatar: `img/avatar-max.jpg`,
        name: `Tex`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the
      unique lightness of Amsterdam. The building is green and from 18th
      century.`,
        time: `June 3019`
      },
      {avatar: `img/avatar-max.jpg`,
        name: `Vex`,
        rating: 1,
        text: `atatatatatatat. The building is green and from 18th
    century.`,
        time: `May 2019`
      },
      {avatar: `img/avatar-max.jpg`,
        name: `Rex`,
        rating: 3,
        text: `A quiet cozy and picturesque that hides behind a a river by the
  unique lightness of Amsterdam. ataaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.`,
        time: `April 2018`
      }
    ]
  },
  {
    id: 2,
    figurePreview: `img/room.jpg`,
    mark: ``,
    price: 20,
    priceText: `night`,
    rating: 5,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    allFigures: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    insideFeatures: [`Apartment`, `3 Bedrooms`, `Max 4 adults`],
    insideOptions: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
      `Cabel TV`,
      `Fridge`
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    user: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    status: ``,
    coordinates: [52.369553943508, 4.85309666406198],
    comments: [
      {avatar: `img/avatar-max.jpg`,
        name: `Vasya`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the
      unique lightness of Amsterdam. The building is green and from 18th
      century.`,
        time: `June 3019`
      },
      {avatar: `img/avatar-max.jpg`,
        name: `Max`,
        rating: 1,
        text: `yiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii. The building is green and from 18th
    century.`,
        time: `May 2019`
      },
      {avatar: `img/avatar-max.jpg`,
        name: `Olga`,
        rating: 3,
        text: `A quiet cozy and picturesque that hides behind a a river by the
  unique lightness of Amsterdam. Yiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii.`,
        time: `April 2018`
      }
    ]
  }
];

it(`<NearOffersList /> sould be render`, () => {
  const tree = renderer
    .create(<NearOffersList places={offers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
