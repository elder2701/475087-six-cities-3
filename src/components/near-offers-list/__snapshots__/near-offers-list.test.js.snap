// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NearOffersList /> sould be render 1`] = `
<div
  className="container"
>
  <section
    className="near-places places"
  >
    <h2
      className="near-places__title"
    >
      Other places in the neighbourhood
    </h2>
    <div
      className="near-places__list places__list"
    >
      <article
        className="cities__place-card place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/room.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                8220
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button place-card__bookmark-button--active button"
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Beautiful & luxurious apartment at great location
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Apartment
          </p>
        </div>
      </article>
      <article
        className="cities__place-card place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/room.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                220
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button place-card__bookmark-button--active button"
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Beautiful & luxurious apartment at great location
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Apartment
          </p>
        </div>
      </article>
      <article
        className="cities__place-card place-card"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/room.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                820
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button place-card__bookmark-button--active button"
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "20%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Beautiful & luxurious apartment at great location
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Apartment
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
